src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleKeyCache.scala

```diff
@@ -178,25 +268,39 @@ class GoogleKeyCache(
       )).compile.drain
     } yield decodedKey
 
-  private def isKeyActive(mostRecentKey: GcsObjectName, serviceAccountKeys: List[ServiceAccountKey]): Boolean = {
-    val keyRetired = System.currentTimeMillis() - mostRecentKey.timeCreated.toEpochMilli > 86400000L * googleServicesConfig.googleKeyCacheConfig.activeKeyMaxAge
-
-    val keyPathPattern(project, petSaEmail, keyId) = mostRecentKey.value
+  /** Cleans up keys by deleting orphans and bringing total key count down to 9.
+    */
+  private def cleanupKeys(pet: PetServiceAccount, cachedKeyObjects: List[CachedKey], serviceAccountKeys: List[ServiceAccountKey]): IO[Unit] = {
+    // if 10 or more keys, delete until we are below 10.
+    val tooManyFunction: IO[_] = if (cachedKeyObjects.size >= 10) {
+      val numToDelete = cachedKeyObjects.size - 9
+      val toDelete = cachedKeyObjects.sortBy(_.timeCreated).takeRight(numToDelete)
+      logger.info(
+        s"cleanupKeys: ${pet.id.project.value}-${pet.serviceAccount.subjectId.value} is reaping ${toDelete.size} keys due to quota"
+      )
+      toDelete.parTraverse(cachedKey => removeKey(pet, cachedKey.keyId))
```

Comment:
`parTraverse` is a bad idea. Google does not alway like concurrent modifications which leads to retry thrashing. Just use `traverse`